agent/discovery/sapsystem.go

```diff
@@ -26,28 +26,28 @@ func (d SAPSystemsDiscovery) GetId() string {
 	return d.id
 }
 
-func (d SAPSystemsDiscovery) Discover() error {
+func (d SAPSystemsDiscovery) Discover() (string, error) {
 	systems, err := sapsystem.NewSAPSystemsList()
 
 	if err != nil {
-		return err
+		return "", err
 	}
 
 	d.SAPSystems = systems
 	for _, s := range d.SAPSystems {
 		err := s.Store(d.discovery.client)
 		if err != nil {
-			return err
+			return "", err
 		}
 
 		// Store SAP System, Landscape and Environment names on hosts metadata
 		err = storeSAPSystemTags(d.discovery.client, s)
 		if err != nil {
-			return err
+			return "", err
 		}
 	}
 
-	return nil
+	return "", nil
```

Comment:
We could put maybe something like `SAP system(s) with ID %s discovered`, and put there all the SAP ids (you will need to pack them in some array in the loop maybe)

Reply:
Changed!